Reed/DefinitionView/viewmodels/DefinitionViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  DefinitionModalViewModel.swift
+//  Reed
+//
+//  Created by Hugo Zhan on 11/3/20.
+//  Copyright © 2020 Roger Luo. All rights reserved.
+//
+
+import SwiftUI
+
+class DefinitionViewModel: ObservableObject {
+    @Published var title = ""
+    @Published var currReading = ""
+    @Published var terms = [Terms]()
+    @Published var definitons = [Defs]()
+    
+    let dictionaryFetcher = DictionaryFetcher()
+    
+    func generateContent(from tappedWord: String) {
+        title = tappedWord
+        var termsArr = [Terms]()
+        var defsArr = [Defs]()
+        let entries = dictionaryFetcher.fetchEntries(of: tappedWord)
+        if entries.endIndex > 0
+            && entries[0].readings[0].terms.endIndex > 0
+            && entries[0].readings[0].terms[0] == tappedWord {
+            currReading = entries[0].readings[0].reading
+        } else {
+            currReading = ""
+        }
+        for entry in entries {
```

Comment:
If there are multiple entries, we should not be appending all of their definitions, readings, and terms. This discussion is here [https://github.com/Rawgers/Reed/pull/25#discussion_r528300597](url).